src/services/node/NodeTransactionPoolService.ts

```diff
@@ -3,20 +3,32 @@ import { INodeTransactionPool } from 'src/types/responses';
 import { AbstractService } from '../AbstractService';
 
 export class NodeTransactionPoolService extends AbstractService {
-	async fetchTransactionPool(): Promise<INodeTransactionPool> {
+	async fetchTransactionPool(
+		includeTip: boolean
+	): Promise<INodeTransactionPool> {
 		const { api } = this;
 
 		const extrinsics = await api.rpc.author.pendingExtrinsics();
 
-		const pool = extrinsics.map((ext) => {
+		if (includeTip) {
```

Comment:
Wondering whether you can avoid the duplication and get away with something like:

```ts
return {
    pool: extrinsics.map((ext) => {
        return {
            hash: ext.hash.toHex(),
            encodedExtrinsic: ext.toHex(),
            tip: includeTip ? ext.tip.toString() : undefined,
        };
    }),
};
```

(afaik undefined fields disappear when serialized to JSON anyway)